### PR TITLE
Auto-increment zxp xml version

### PR DESCRIPTION
**Pull request by ccaillot**
## Changelog Description
Add checksums generating & checksums comparing process in order to detect changes in host's extension folder. If changes detected, we bump the xml version before to generate  zxp file.

### CODE REVIEW

File: tools/zxp/generate_zxp.ps1 [hunk 1]
```diff
@@ -19,4 +93,17 @@ foreach ($CURR_HOST in $HOSTS) {
 
     # Generate and sign the ZXP file with the OpenPype certificate
     & $PATH_ZXP_SIGN_SOFTWARE -sign $HOST_ZXP_SOURCE $HOST_ZXP_DEST $PATH_ZXP_CERTIFICATE OpenPype
```
Comment (BenSouchet) [inline tools/zxp/generate_zxp.ps1:95]: You need to generate the ZXP after bumping the version not before, else you will have the wrong version number on the generated ZXP.
Move this operation/line at the end after the bump of the version and the new checksum generation

**Review by BenSouchet — CHANGES_REQUESTED** [review]

File: tools/zxp/generate_zxp.ps1 (hunk 1 above)
Comment (ccaillot) [reply tools/zxp/generate_zxp.ps1:95]: Effectivement, je corrige ça

File: tools/zxp/generate_zxp.ps1 [hunk 2]
```diff
@@ -1,3 +1,64 @@
...
+            $checksum_valid = $false
+        }
+    }
+
+    # Check for missing files
+    foreach ($checksum_relative_path in $checksums_dict.Keys) {
+        $checksum_path = Join-Path -Path $folder_to_compare -ChildPath $checksum_relative_path.Replace("/", "\")
+        if (-not (Test-Path -Path $checksum_path)) {
+            Write-Host "File missing: $checksum_path"
+            $checksum_valid = $true
```
[the remark this answers is not in the document]
Comment (ccaillot) [reply tools/zxp/generate_zxp.ps1]: Effectivement, ce type de cas m'avait échappé, je viens de release le fix

**Review by BenSouchet — APPROVED** [review]
C'est good pour moi, tu peux l'ajouter à la future release `3.16.9-quad-1.16.0`

**Pull request merged**